### PR TITLE
`Development`: Improve selectors used in e2e cypress tests for exam dates

### PR DESCRIPTION
**Pull request by rs-maker**
### Motivation and Context
 We are using selectors in Cypress tests that are not ids. These selectors might change during development and we therefore want to use more stable id selectors.
In this PR i fix all the bad selectors used in Exam date verification specs and related page objects by adding ids where needed. I also remove all assertions asserting notificiations as that is not an important part of cypress testing.

### Steps for Testing

We have a tutorial for testing cypress PRs <a href="https://github.com/ls1intum/Artemis/blob/develop/src/test/cypress/README.md"> here </a>

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/test/cypress/integration/exam/ExamDateVerification.spec.ts
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamDateVerification.spec.ts:130]: Here it might make sense to give the element an id and then only assert that the exam title is shown in it.
Something like this
```suggestion
                        cy.get('#exam-finished-message', { timeout: 40000 }).should('contain.text', exam.title);
```

**Review by jhainzlschmid — CHANGES_REQUESTED** [review]
One minor thing where we could reduce the dependency on Artemis translations. There is also an unused variable warning, so you should remove that one. I saw that one  page object still contains a css selector. However this selector is refactored in #4533 so I suggest we wait until this PR has been merged (should be very soon). Otherwise the changes look good to me 👍

**Review by jhainzlschmid — DISMISSED** [review]
LGTM 👍

**Review by jhainzlschmid — CHANGES_REQUESTED** [review]
One test in this spec failed in the [bamboo buildplan](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-723). The issue seems to be that the test expects to see the end screen of the exam, but the exam is still running for more than 20 seconds if the build agent is fast. This causes the cypress to timeout before the exam ends. Either add a `cy.wait` to wait until the end of the exam or give cypress more time in that one assertion

**Comment by rs-maker** [comment]
Added a wait so fast or slow build agents should not fail anymore. 
Tests <a href="https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-730"> pass again </a>

**Pull request merged**
